### PR TITLE
required_audience should be a list in order to support enterprise oau…

### PR DESCRIPTION
**Pull request by m4rkw**
…th2 deployments where the claim is different for kibana access vs direct api access.

### Description
[Describe what this change achieves]
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
* Why these changes are required?
* What is the old behavior before changes and new behavior after changes?

### Issues Resolved

- https://github.com/opensearch-project/security/issues/3723

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java
Comment (cwperks) [inline src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java:75]: wdyt about keeping this as a `List<String>` and avoid having to re-cast this back to a list in the `JwtVerifier`?

**Review by cwperks — COMMENTED** [review]
Thank you for the contribution @m4rkw! This change looks good to me, but I'd like to see a test that verifies the intended behavior. [JwtAuthenticationTests](https://github.com/opensearch-project/security/blob/main/src/integrationTest/java/org/opensearch/security/http/JwtAuthenticationTests.java) or [HttpJwtAuthenticatorTest](https://github.com/opensearch-project/security/blob/main/src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java#L462-L483) may be a good locations for a test case to assert the behavior. Can you add a test for this change?

**Comment by m4rkw** [comment]
Hey

I'm happy for you to change this however you like, I'm not a java developer so have no idea if its sensible/efficient/anything else. I just had to figure out how to patch it in our environment, I wouldn't know where to start with writing tests.

Cheers,
Mark

File: src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java
Comment (DarshitChanpura) [reply src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java:75]: +1. Is there any specific reason why Object array was considered for required_audience over string list?

File: src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/JwtVerifier.java
Comment (DarshitChanpura) [inline src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/JwtVerifier.java:137]: Thoughts on this, once required_audience is set to string list:
```suggestion
        if (requiredAudience.isEmpty() || !requiredAudience.contains(audience)) {
                throw new BadJWTException("Invalid audience");
            }
```

File: src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/JwtVerifier.java
Comment (DarshitChanpura) [inline src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/JwtVerifier.java:131]: this: should be modified as well since claims can now have multiple audiences

**Review by peternied — CHANGES_REQUESTED** [review]
Thanks for the contribution @m4rkw This outstanding feedback needs to be addressed before this change is merged - can you address it or would you rather close this PR?

**Comment by m4rkw** [comment]
> Thanks for the contribution @m4rkw This outstanding feedback needs to be addressed before this change is merged - can you address it or would you rather close this PR?

Sorry I'm confused, I replied earlier, what else do you need?

**Comment by m4rkw** [comment]
ah wait no I replied on the issue here: https://github.com/opensearch-project/security/issues/3723#issuecomment-1842458659

**Comment by peternied** [comment]
@m4rkw From your comment it sounds like you aren't planning on working on this change anymore I'm going to close out this pull request.  It will be associated with the issue in case someone else comes along with looking to build off of what you've done.

If you change your mind and invest in contributing to the project free free to reopen if you re-engage on this issue or others in the project, thanks.

**Comment by m4rkw** [comment]
@peternied I'm not a java developer so I'm not really in a good position to flesh this out to the point where it could be merged. I reported the lack of support for multiple audiences in the related issue and mentioned that we had patched it internally in a "just make it work" fashion, and it was suggested on the issue that I should raise a PR so I did that with our internal patch.

Unfortunately I don't really have the time to learn java and its test frameworks at the moment in order to flesh this out and make it into something that could be merged, I was rather hoping the core developers of the project might be able to do that. It sounds like that's not the case for the moment.

**Pull request closed without merging**
